### PR TITLE
Added display hits ranking scores [v1.3]

### LINKED ISSUE #459: [v1.3] Display hits ranking scores
⚠️ This issue is generated, it means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Sorry if this is already wholly/partially implemented. Feel free to let me know about the state of this issue in the repo.

Related to https://github.com/meilisearch/integration-guides/issues/280

----

This issue is divided into two sections, first, you need to make the implementation, and second, you must update the code-samples (no one likes outdated docs, right?).

### New implementation

Related to:
- issue: https://github.com/meilisearch/meilisearch/issues/3771
- discussion: https://github.com/meilisearch/product/discussions/594 
- spec: https://github.com/meilisearch/specifications/pull/252

This feature aims to return ranking details for each document to understand and tweak the score of the documents more efficiently.

Ensure the SDKs can handle the new search parameter `showRankingScore`. Also, ensure the SDK can handle the `_rankingScore` attribute in the matched `hits`.

### Code samples

Inside of this file: `.code-samples.meilisearch.yml`:

1. Create a new entry with this key `search_parameter_guide_show_ranking_score_1` containing a call to the `search` method using the new behavior from the index `movies` using `showRankingScore=true` and a `q` with `'dragon'`.
Add a new code sample to the `.code-samples.meilisearch.yml` file:

Use this as a reference _if the previous description was not helpful_:

```
search_parameter_guide_show_ranking_score_1: |-
  POST 'http://localhost:7700/indexes/movies/search' 
  with data: { 
    "q": "dragon", "showRankingScore": true
  }
```

TODO:
- [ ] Add the ability receive a new param in the `search` method called `showRankingScore`.
- [ ] Add the ability handle the new `_rankingScore` key/value in the search hits' response.
- [ ] Update the code-samples accordingly
- [ ] Add integration tests

### PR DESCRIPTION
**Pull request by andre-m-dev**
# Pull Request

## Related issue
Fixes #459

## What does this PR do?
- Ensure the SDKs can handle the new search parameter showRankingScore

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: spec/meilisearch/index/search/show_ranking_score_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+RSpec.describe 'MeiliSearch::Index - Search with ranking score' do
+  include_context 'search books with genre'
+
+  it 'shows the ranking score when showRankingScore is true' do
+    response = index.search('hobbit', { showRankingScore: true })
```
Comment (brunoocasali) [inline spec/meilisearch/index/search/show_ranking_score_spec.rb:7]:
```suggestion
    response = index.search('hobbit', { show_ranking_score: true })
```

File: lib/meilisearch/index.rb [hunk 2]
```diff
@@ -220,6 +220,8 @@ def delete_all_documents!
 
     ### SEARCH
 
+    # options: A Hash
+    #   showRankingScore - To see the ranking scores for returned documents
```
Comment (brunoocasali) [inline lib/meilisearch/index.rb:224]:
```suggestion
    #   show_ranking_score - To see the ranking scores for returned documents
```

File: spec/meilisearch/index/search/show_ranking_score_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    expect(response['hits'][0]).to have_key('_rankingScore')
+  end
+
+  it 'hides the ranking score when showRankingScore is false' do
+    response = index.search('hobbit', { showRankingScore: false })
```
Comment (brunoocasali) [inline spec/meilisearch/index/search/show_ranking_score_spec.rb:12]:
```suggestion
    response = index.search('hobbit', { show_ranking_score: false })
```

**Review by brunoocasali — COMMENTED** [review]
Can you make the code snake_case? The Ruby SDK can support that :)

Thanks for contributing!

**Comment by andre-m-dev** [comment]
@brunoocasali thanks for the advice

**Review by brunoocasali — APPROVED** [review]
bors merge

**Pull request closed without merging**